src/ioctls/device.rs

```diff
@@ -117,7 +158,7 @@ impl DeviceFd {
     /// let vm = kvm.create_vm().unwrap();
     ///
     /// // As on x86_64, `get_device_attr` is not necessarily needed. Therefore here
-    /// // the code example is only for AArch64.
+    /// // the code example is only for AArch64 and RISC-V 64-bit.
     /// #[cfg(any(target_arch = "aarch64"))]
```

Comment:
Mh, I admittedly think that for doc comments, having the doc comments only for a single architecture is fine. I'd move the riscv64 example you add below to the unittests instead (also, I now realize that the `cfg`s are _already_ shown on doc.rs. ugh. disregard my previous comment about that, maybe I'll post a separate fixup for that later

Reply:
Oh, I just recollected, I'm writing these doc examples because without completing them, the doc test simply wouldn't pass on `riscv64` architecture :(

Reply:
Huh, can you give an example? I don't immediately see why they wouldn't pass

Reply:
Yes, I'll illustrate per commit:

- 6a2d5dc: `type_` field is architecture specific, when running on riscv the field would be missing then failing
- a75b454: fails on riscv because 500 are too big
- 8fc3cda: fails because VFIO is not yet available on riscv
- ea0e5a4: this one is fine, because it does nothing on riscv
- 3649c68: `set_gsi_routing` requires AIA created in advance on riscv
- 2246f1c: `asm_code` is architecture specific

Reply:
- https://github.com/rust-vmm/kvm-ioctls/commit/6a2d5dcb80b273a52a040d4918a55e69b15177dd: this one is fine, its just a single line
- https://github.com/rust-vmm/kvm-ioctls/commit/a75b454cba25cebf487b672e082601f0cd130841: Have the example only be the ARM64 variant, and do the riscv check in unittests
- https://github.com/rust-vmm/kvm-ioctls/commit/8fc3cdad44d2b74f7eb5ae21fe4a91ee6551b8d5: cfg it to not run on riscv, do something riscv related in unittests. If the QEMU patches get merged, we can drop the cfg
- https://github.com/rust-vmm/kvm-ioctls/commit/3649c684e2552872b332856630f1e5d490a1a05f: This one is again fine
- https://github.com/rust-vmm/kvm-ioctls/commit/2246f1c7965538823093fc1a328d907e45cd1ad7: also fine

I think I'm drawing the line at: single line additions because setup for ioctls is different on different architectures (fine) vs duplication of large parts of/the entire example only to pass different constants to the ioctl (let's not do this).